src/app/api/teams/overview/route.ts

```diff
@@ -45,44 +48,3 @@ export async function GET(req: NextRequest) {
     });
   }
 }
-
-function getIntervalForDate(date: Date, scoreInterval: any[]) {
-  for (const interval of scoreInterval) {
-    const startDate = new Date(interval.sTime);
-    const endDate = new Date(interval.eTime);
-    if (startDate < date && endDate > date) {
-      return interval.value;
-    }
-  }
-  return 0;
-}
-
-function mergeIntervals(userContributors: any[]): any[] {
-  const map = new Map();
-  userContributors.forEach((userContributor) => {
-    userContributor.score_interval.forEach((interval: any) => {
-      const intervalRow = map.get(interval.eTime);
-      if (intervalRow) {
-        intervalRow.value = intervalRow.value + interval.value;
-        map.set(interval.eTime, intervalRow);
-      } else {
-        map.set(interval.eTime, interval);
-      }
-    });
-  });
-  return Array.from(map.values());
-}
-
-function calculatePercentageDifference(value1: number, value2: number): number {
-  const difference: number = value2 - value1;
-  const percentageDifference: number = (difference / value1) * 100;
-  return percentageDifference;
-}
-
-function sumScore(teamContributors: TeamContributorDto[]) {
-  let score = 0;
-  teamContributors.forEach((teamContributor) => {
-    score += Number(teamContributor.contributionScore);
```

Comment:
isn't this used on the teamDetails page?
I didn't saw you changing the api being used there, can you explain it? As far as I understood you are merging it and leaving it only on the /overview api route. but I didn't saw how you parametrize it to also use it for a specific teamID, neither how you are calling that one now instead of this previous one on the teamId page

Reply:
I move this helper function out to the utils class,  we use the same functions for both insights cards